transcom/mymove
# Int b 20766 filename correct on downloads

## [B-20766](Downloading Docs File Names - URL generation updates)

## Summary
Files downloaded from S3 had filenames matching the UUID of the file in the storage bucket instead of the original filename. This was due to how the browser handles cross-origin downloads (`<a href="..." download="filenameppdf">` no longer works, a header attribute is expected in the content-disposition)

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### How to test

Enable s3 local testing per link here https://dp3.atlassian.net/wiki/spaces/MT/pages/1470955567/How+to+test+storing+data+in+S3+locally

Run server:
aws-vault exec transcom-gov-dev -- make server_run

1. Access the app as a services_counselor
2. Select a move and click "Supporting Documents" tab
3. Upload a new document
4. Refresh page after ~10 seconds (avoid another bug - outside of scope)
5. Click "Download file" once file appears within the viewer (refresh again every ~10 seconds until it does appear)
6. Verify that the filename matches that displayed in the Uploads table (not a uuid value)

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots

**UUID filename (before)**
<img width="1726" alt="image" src="https://github.com/user-attachments/assets/b3e2f60a-7e49-471c-b556-7050a70bb31f">
